var/spack/repos/builtin/packages/intel-oneapi-compilers/package.py

```diff
@@ -518,6 +518,27 @@ def extend_config_flags(self):
         classic_flags = ["-gcc-name={}".format(self.compiler.cc)]
         classic_flags.append("-gxx-name={}".format(self.compiler.cxx))
 
+        # On ubuntu systems `icc` will only add `/usr/include/x86_64-linux-gnu` to the include
+        # directories if the `gcc` that `icc` was installed with also reports `x86_64-linux-gnu` as
+        # its install target. By default `gcc` installed with Spack uses `config.guess` to
+        # determine the install target. For ubuntu systems `config.guess` returns
+        # `x86_64-pc-linux-gnu` instead of `x86_64-linux-gnu`.
+        # `self.compiler.cc` is gcc, so we can assume `-print-multiarch` is a valid option. In
+        # order to get the multiarch include directory used by `gcc` we need to ask `cc1` with the
+        # correct flags.
+        gcc = Executable(self.compiler.cc)
+        cc1 = Executable(gcc("-print-prog-name=cc1", output=str).strip())
+        multiarch_triplet = gcc("-print-multiarch", output=str).strip()
```

Comment:
`multiarch_triplet` can be empty string, then `i.endswith("\n")` is always true.

Reply:
I have added a change so that the `incdir` is only going to be created if `multiarch_triplet` is not empty.